### DISCUSSION TITLE
How to set size constraints in gui?

### DISCUSSION THREAD

**Discussion by mrbogus** [isl-org/Open3D#5057] [Q&A]
Hi,

I just try to create a GUI using the gui framework. What I want to achieve is to get a window with a few buttons (and possibly some more widgets) at the top and bottom of the window and two images in the middle. The two images should be the same size and next to each other. I cannot figure out how to create a layout that makes this work. I come from the QT world, so there i simply would set the preferred size of the two images to be half of the window width. Any hint how I can achieve this in Open3d?

Regards
Matthias

**Comment by forrestjgq** [comment]
```
Vert()
    buttons inside a horiz, better no list contained in it
    stretch
    Horiz(stretch, image, image, stretch)
    stretch
    fixe(with height of button)
```

**Reply by mrbogus** [reply]
Thanks for the suggestion. That is what I already tried. It works, until I actually load an image into the widget (be it while while creation or afterwards via a button click). As soon I do this, the image is loaded with the full size of the image file and that crashes the whole layout instead of limiting the ImageWidget size to the constraints of the window.

_edit_: Here is a short gif of the behavior: ![imagewidget_error](https://user-images.githubusercontent.com/5362233/165766271-0b86c831-479c-44f1-9dce-828b184d10a3.gif)
